pkg/apphealth/status.go

```diff
@@ -0,0 +1,31 @@
+package apphealth
+
+import "time"
+
+const (
+	AppStatusUnhealthy uint8 = 0
+	AppStatusHealthy   uint8 = 1
+
+	// reportMinInterval is the minimum interval between health reports.
+	reportMinInterval = time.Second
+)
+
+type Status struct {
+	State    uint8   `json:"state"`
+	TimeUnix int64   `json:"timeUnix"`
+	Reason   *string `json:"reason,omitempty"`
+}
+
+func (s *Status) IsHealthy() bool {
+	return s.State == AppStatusHealthy
+}
+
+// DefaultStatus returns a default status for the app.
+// This defaults to healthy so subscriptions can start right away.
+func DefaultStatus() *Status {
+	return &Status{
+		State:    AppStatusHealthy,
```

Comment:
Should this not be false be default?

Reply:
no, there is a note somewhere in the code that it should default to healthy so subscriptions start up right away, so left as is to not bring in random changes, and just keep it a clean refactor :)

Reply:
actually yeah i'll leave defaults alone and make sure the part accounting for this sets it to healthy